internal/transformer/classes/generators/dockerfilegenerators/net5appdockerfilegenerator.go

```diff
@@ -0,0 +1,249 @@
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		if a.Artifact != artifacts.ServiceArtifactType {
+			continue
+		}
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var ports []int32
+		var net5Config Net5TemplateConfig
+		net5Config.AppName = a.Name
+		net5Config.CsprojFilePath, err = filepath.Rel(a.Paths[artifacts.ProjectPathPathType][0], a.Paths[CsprojFilePathType][0])
+		if err != nil {
+			logrus.Errorf("Error while getting the relative path of csproj file%s", err)
+		}
+		jsonFiles, err := common.GetFilesByExt(a.Paths[artifacts.ProjectPathPathType][0], []string{".json"})
+		if err != nil {
+			logrus.Errorf("Error while finding json files %s", err)
+		}
+		for _, jsonFile := range jsonFiles {
+			if filepath.Base(jsonFile) == launchSettingsJSON {
+				launchSettings := LaunchSettings{}
+				if err := common.ReadJSON(jsonFile, &launchSettings); err != nil {
+					logrus.Errorf("unable to read the launchSettings.json file: %s", err)
+					continue
+				}
+				if launchSettings.Profiles[net5Config.AppName] != nil {
+					profiles := launchSettings.Profiles[net5Config.AppName].(map[string]interface{})
+					applicationUrls := profiles["applicationUrl"].(string)
+					Urls := strings.Split(applicationUrls, ";")
+					re := regexp.MustCompile("[0-9]+")
+					for _, url := range Urls {
+						re1 := regexp.MustCompile("^https://")
+						if len(re1.FindAllString(url, -1)) != 0 && re1.FindAllString(url, -1)[0] == "https://" {
+							portStr := re.FindAllString(url, 1)[0]
+							port, err := strconv.ParseInt(portStr, 10, 32)
+							if err != nil {
+								logrus.Errorf("Error while converting the port from string to int : %s", err)
+							}
+							net5Config.HTTPSPort = int32(port)
+						}
+						re2 := regexp.MustCompile("^http://")
+						if len(re2.FindAllString(url, -1)) != 0 && re2.FindAllString(url, -1)[0] == "http://" {
+							portStr := re.FindAllString(url, 1)[0]
+							port, err := strconv.ParseInt(portStr, 10, 32)
+							if err != nil {
+								logrus.Errorf("Error while converting the port from string to int : %s", err)
+							}
+							net5Config.HTTPPort = int32(port)
+						}
```

Comment:
Why not use a single regex to compute it all?

Reply:
Yes sure, I have made the changes in the code.